### DISCUSSION TITLE
REQ: Integrate camquake (or equivalent camera scripting)

### DISCUSSION THREAD

**Discussion by meag** [ezQuake/ezquake-source#515] [Ideas]
Website: http://jogi.netdome.biz/camquake/index.html
Source: https://github.com/jogi1/camquake

Had a brief look, hooks with the existing game loop etc (potential starting points) seem to be:

cl_main: MOSDM_Init() in CL_Init()
cl_screen: MOSDM_Screen_Hook() in SCR_DrawElements() [between hud_editor and democontrols]
gl_rmain: MVD_Cam_Render_Hook/CamSys_Handler/MOSDM_Change_View() in R_RenderScene() (before R_SetupFrame)
gl_rmain: MVD_Cam_Render_Beziers() in R_RenderView() after DrawViewModel

There may well be others.

**Comment by blaps** [comment]
This!

**Comment by meag** [comment]
https://github.com/jogi1/ezquake-source/commit/2425e6478a76360888f30d0da9c1c2ca798f069e#diff-6230f6cc337efd1da5593eafeb714c5eeb4c4d7106dfbd4490d26423d1f55ed8

Some inline OpenGL calls, so currently immediate-mode only

**Comment by etaxirunized97-jpg** [comment]
Ein kurzer Blick zeigt, dass mögliche Startpunkte Hooks im bestehenden https://de-crocoslots.casino/ Game-Loop sein könnten: cl_main: MOSDM_Init() in CL_Init(). cl_screen: MOSDM_Screen_Hook() in SCR_DrawElements() (zwischen hud_editor und democontrols). gl_rmain: MVD_Cam_Render_Hook / CamSys_Handler / MOSDM_Change_View() in R_RenderScene() (vor R_SetupFrame) und MVD_Cam_Render_Beziers() in R_RenderView() nach DrawViewMo._
